### PR TITLE
games-action/minetest: Add prometheus client support

### PR DESCRIPTION
**Pull request by vilhelmgray**

### CODE REVIEW

**Comment by zmedico** [comment]
> ## Pull Request assignment
> **www-servers/civetweb**: @zmedico

Merged in fc7e00f0413548a07650f8467f293c2a8cb996a8.

File: dev-cpp/benchmark/benchmark-1.5.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+SRC_URI="https://github.com/google/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="Apache-2.0"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
+RESTRICT="!test? ( test )"
+
+RDEPEND=""
+DEPEND="${RDEPEND}"
```
Comment (juippis) [inline dev-cpp/benchmark/benchmark-1.5.1.ebuild:19]: This seems a bit useless.

File: dev-cpp/benchmark/metadata.xml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+	<!-- maintainer-needed -->
```
Comment (juippis) [inline dev-cpp/benchmark/metadata.xml:4]: We can't add new packages without maintainers, it's against QA policy. 
https://projects.gentoo.org/qa/policy-guide/maintainer.html#pg0602

File: dev-cpp/prometheus-cpp/metadata.xml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+	<longdescription lang="en">
+		This library aims to enable Metrics-Driven Development for C++ services.
+		It implements the Prometheus Data Model, a powerful abstraction on which
+		to collect and expose metrics. We offer the possibility for metrics to
+		be collected by Prometheus, but other push/pull collections can be added
+		as plugins.
+	</longdescription>
+	<!-- maintainer-needed -->
```
Comment (juippis) [inline dev-cpp/prometheus-cpp/metadata.xml:11]: Same here.

File: dev-cpp/prometheus-cpp/prometheus-cpp-0.9.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+inherit cmake
+
+DESCRIPTION="Prometheus Client Library for Modern C++"
+HOMEPAGE="https://github.com/jupp0r/prometheus-cpp"
+SRC_URI="https://github.com/jupp0r/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="+pull +push test zlib"
```
Comment (juippis) [inline dev-cpp/prometheus-cpp/prometheus-cpp-0.9.0.ebuild:15]: These aren't very descriptive... What do they actually do? Should they be enabled unconditionally instead?

File: dev-cpp/prometheus-cpp/prometheus-cpp-0.9.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+RESTRICT="!test? ( test )"
+
+RDEPEND="
+	pull? (
+		www-servers/civetweb[cxx]
+		dev-libs/cppcodec
+		sys-libs/zlib
```
Comment (juippis) [inline dev-cpp/prometheus-cpp/prometheus-cpp-0.9.0.ebuild:22]: Sort.

File: dev-cpp/prometheus-cpp/prometheus-cpp-0.9.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+DEPEND="${RDEPEND}
+	test? (
+		dev-cpp/benchmark
+		dev-cpp/gtest
+	)"
+
+src_configure() {
+	local mycmakeargs=(
+		-DENABLE_PULL=$(usex pull)
+		-DENABLE_PULL=$(usex push)
```
Comment (juippis) [inline dev-cpp/prometheus-cpp/prometheus-cpp-0.9.0.ebuild:34]: You have `PULL` twice.

File: dev-cpp/prometheus-cpp/prometheus-cpp-0.9.0.ebuild (hunk 4 above)
Comment (vilhelmgray) [reply dev-cpp/prometheus-cpp/prometheus-cpp-0.9.0.ebuild:15]: `pull` will build the `libprometheus-cpp-pull.so.0.9.0` library, while `push` will build the `libprometheus-cpp-push.so.0.9.0` library;  these respectively enable clients to pull and push data from/to a prometheus server. These two libraries seem to be relatively small (36K and 76K respectively) so I don't think it'll be that bad to just build them unconditionally. I'll remove these flags then.

**Comment by vilhelmgray** [comment]
> ```
> 46/61 Test #60: run_state_assembly_test_CHECK ..............***Failed    0.01 sec
> 
> 98% tests passed, 1 tests failed out of 61
> 
> Total Test time (real) =   3.04 sec
> 
> The following tests FAILED:
> 	 60 - run_state_assembly_test_CHECK (Failed)
> Errors while running CTest
>  * ERROR: dev-cpp/benchmark-1.5.1::testworld failed (test phase):
>  *   Tests failed. When you file a bug, please attach the following file: 
>  * 	/var/tmp/portage/dev-cpp/benchmark-1.5.1/work/benchmark-1.5.1_build/Testing/Temporary/LastTest.log
> ```
> 
> You will most likely receive a bug report about it soon.

Looks like this is testing the assembly code produced by the compiler. This might be an upstream so I'll open a issue page about this with them.

> ```
> >>> Test phase: dev-cpp/prometheus-cpp-0.9.0
>  * Working in BUILD_DIR: "/var/tmp/portage/dev-cpp/prometheus-cpp-0.9.0/work/prometheus-cpp-0.9.0_build"
> ctest -j 16 --test-load 16
> Test project /var/tmp/portage/dev-cpp/prometheus-cpp-0.9.0/work/prometheus-cpp-0.9.0_build
>     Start 1: prometheus_test
>     Start 2: benchmarks
> 1/2 Test #1: prometheus_test ..................***Failed    1.79 sec
> 2/2 Test #2: benchmarks .......................   Passed  129.71 sec
> 
> 50% tests passed, 1 tests failed out of 2
> 
> Label Time Summary:
> Benchmark    = 129.71 sec*proc (1 test)
> 
> Total Test time (real) = 129.71 sec
> 
> The following tests FAILED:
> 	  1 - prometheus_test (Failed)
> Errors while running CTest
>  * ERROR: dev-cpp/prometheus-cpp-0.9.0::testworld failed (test phase):
>  *   Tests failed. When you file a bug, please attach the following file: 
>  * 	/var/tmp/portage/dev-cpp/prometheus-cpp-0.9.0/work/prometheus-cpp-0.9.0_build/Testing/Temporary/LastTest.log
> ```

I'm unable to reproduce this failure. I get the following:
```
>>> Test phase: dev-cpp/prometheus-cpp-0.9.0
 * Working in BUILD_DIR: "/var/tmp/portage/dev-cpp/prometheus-cpp-0.9.0/work/prometheus-cpp-0.9.0_build"
ctest -j 5 --test-load 999
Test project /var/tmp/portage/dev-cpp/prometheus-cpp-0.9.0/work/prometheus-cpp-0.9.0_build
    Start 1: prometheus_test
    Start 2: benchmarks
1/2 Test #1: prometheus_test ..................   Passed    2.21 sec
2/2 Test #2: benchmarks .......................   Passed  136.15 sec

100% tests passed, 0 tests failed out of 2

Label Time Summary:
Benchmark    = 136.15 sec*proc (1 test)

Total Test time (real) = 136.15 sec
 * Tests succeeded.
>>> Completed testing dev-cpp/prometheus-cpp-0.9.0
```

How are you executing this test? Maybe we have some difference in USE flags?

> ```
>  * Messages for package dev-cpp/prometheus-cpp-0.9.0:
> 
>  * One or more CMake variables were not used by the project:
>  *   THIRDPARTY_CIVETWEB_WITH_SSL
> ```
> 
> I can't test minetest due to a bug in dependency,
> https://bugs.gentoo.org/736342 but please do check the cmake variable being correct (is there a typo?) while we wait for this to be fixed.

This CMake variable is only used if we are compiling with thirdparty libraries (we're not) so I've removed it in my latest force-push.

**Comment by vilhelmgray** [comment]
Upstream benchmark Github issue for reference: https://github.com/google/benchmark/issues/1020

**Comment by juippis** [comment]
Can we fix irrlicht dependency / irrlicht package here to get this moving forward? You probably have a patch for either case?

**Comment by vilhelmgray** [comment]
Here's a PR to resolve the Irrlicht issue: https://github.com/gentoo/gentoo/pull/17426

Regarding the benchmark issues, upstream believes those assembly tests are not valid and suggests disabling them for now with `-DBENCHMARK_ENABLE_ASSEMBLY_TESTS=no`, so I'll make that change.

**Review by juippis — APPROVED** [review]

**Pull request closed without merging**